clarity/contracts/arkadiko-oracle-v2-4.clar

```diff
@@ -0,0 +1,252 @@
+        )
+      )
+    )
+  )
+)
+
+;; ---------------------------------------------------------
+;; Price Helpers
+;; ---------------------------------------------------------
+
+(define-public (get-pyth-info (token (string-ascii 12)))
+  (let (
+    (price-info (map-get? prices-pyth { token: token }))
+  )
+    (if (is-some price-info)
+      (let (
+        (price (try! (get-price-pyth (get feed-id (unwrap-panic price-info)))))
+        (decimals (get decimals (unwrap-panic price-info)))
+      )
+        (ok { last-price: price, last-block: burn-block-height, decimals: decimals })
+      )
+      (err ERR-PRICE-NOT-FOUND)
+    )
+  )
+)
+
+(define-public (get-dia-info (token (string-ascii 12)))
+  (let (
+    (price-info (map-get? prices-dia { token: token }))
+  )
+    (if (is-some price-info)
+      (let (
+        (price (unwrap-panic (get-price-dia (get key (unwrap-panic price-info)))))
+        (decimals (get decimals (unwrap-panic price-info)))
+      )
+        (ok { last-price: price, last-block: burn-block-height, decimals: decimals })
+      )
+      (err ERR-PRICE-NOT-FOUND)
+    )
+  )
+)
+
+(define-public (get-custom-info (token (string-ascii 12)))
+  (let (
+    (price-info (map-get? prices-custom { token: token }))
+  )
+    (if (is-some price-info)
+      (let (
+        (price (try! (get-price-custom (get key (unwrap-panic price-info)))))
+        (decimals (get decimals (unwrap-panic price-info)))
+      )
+        (ok { last-price: price, last-block: burn-block-height, decimals: decimals })
+      )
+      (err ERR-PRICE-NOT-FOUND)
+    )
+  )
+)
+
+(define-public (get-manual-info (token (string-ascii 12)))
+  (let (
+    (price-info (map-get? prices-manual { token: token }))
+  )
+    (if (is-some price-info)
+      (let (
+        (price (get price (unwrap-panic price-info)))
+        (decimals (get decimals (unwrap-panic price-info)))
+      )
+        (ok { last-price: price, last-block: burn-block-height, decimals: decimals })
+      )
+      (err ERR-PRICE-NOT-FOUND)
+    )
+  )
+)
+
+
+;; ---------------------------------------------------------
+;; Oracles
+;; ---------------------------------------------------------
+
+(define-public (get-price-pyth (price-feed-id (buff 32)))
+  (let (
+
+    ;; TODO: Update for mainnet
+    ;; (storage 'SP3R4F6C1J3JQWWCVZ3S7FRRYPMYG6ZW6RZK31FXY.pyth-storage-v3)
+    ;; (oracle-data (try! (contract-call? 'SP3R4F6C1J3JQWWCVZ3S7FRRYPMYG6ZW6RZK31FXY.pyth-oracle-v3 get-price price-feed-id storage)))
+
+    (storage .pyth-storage-v3)
+    (oracle-data (try! (contract-call? .pyth-oracle-v3 get-price price-feed-id storage)))
+  )
+    (ok (to-uint (get price oracle-data)))
+  )
+)
+
+(define-public (get-price-dia (key (string-ascii 32)))
+  (let (
+    ;; TODO: Update for mainnet
+    ;; (oracle-data (unwrap-panic (contract-call? 'SP1G48FZ4Y7JY8G2Z0N51QTCYGBQ6F4J43J77BQC0.dia-oracle get-value key)))
+    (oracle-data (unwrap-panic (contract-call? .dia-oracle get-value key)))
```

Comment:
Q: do we have a list of the keys for DIA, and also for Pyth which `price-feed-id` values we can pass? would be good to keep them as comments here perhaps